GDJS/Runtime/events-tools/soundtools.ts

```diff
@@ -428,6 +428,28 @@ namespace gdjs {
         runtimeScene
           .getSoundManager()
           .unloadAudio(soundFile, /* isMusic= */ true);
+
+      export const FadeVolume = (
+        runtimeScene: gdjs.RuntimeScene,
+        channel: integer,
+        toVolume: float,
+        timeOfFade: float
+      ) => {
+        //Clamping volume
+        if (toVolume > 100) {
+          toVolume = 100;
+        }
+        if (toVolume < 0) {
+          toVolume = 0;
+        }
```

Comment:
Don't do your own clamping, and don't do it here!

-> We already have a clamping function at the top of howler-sound-manager that you can use
-> Functions in events tools are supposed to be just functional wrappers to the object oriented APIs, as events are functional. Any logic relevant to the functionality of the API should be present on the object oriented function, in this case `HowlerSound.fade`.

Reply:
Oh I didn't realize that, I wrote my comment too quickly. Indeed, @Oxey405 your fade method should be implemented in our class `HowlerSoundManager` and you can use `clampVolume` that is defined in the same file.

And as @arthuro555 mentioned it, we should make sure that the `HowlerSound._volume` corresponds to the `Howl` instance `volume` property.

Reply:
> your fade method should be implemented in our class HowlerSoundManager

I disagree, it is already implemented on `HowlerSound`, and it isn't the responsibility of the manager to call methods on the sounds. What I was saying is that the lamping, which is a necessity for the API to work correctly, should be present on the actual API (`HowlerSound.fade`) and not on the events-tools "bridge function". See the volume action for example, we are still calling the method on the `HowlerSound`, but we are not clamping in events-tools: we are clamping in `HowlerSound.setVolume`.

> we should make sure that the `HowlerSound._volume` corresponds to the Howl instance volume property

So I checked thee code again, and we shouldn't set the `_volume` in the fade method. It is there because the Howl doesn't has to have been preloaded, and it will need some time to load before playing. Until then, we cannot access the `Howl` to set the volume or other properties. Therefore, volume and other properties are stored on the `HowlerSound` instance until the Howl is ready, and we can pass on those properties to the Howl. 
Therefore, we should keep the fade settings on a separate member object, and [when starting to play the sound](https://github.com/4ian/GDevelop/blob/9710360180ab41461f95d1fc6b2c36ca04bdb44e/GDJS/Runtime/howler-sound-manager/howler-sound-manager.ts#L107), if a fade was requested for that sound before it was loaded, then the fade should be executed with the stored parameters (after setting the volume, since that is what defines the beginning of the fade).

To ensure that `getVolume` returns the right volume, we should only return `_volume` while the sound is not loaded (`_id === null`), then get the volume from the howl once it started playing.

Reply:
Yes sorry, `HowlerSoundManager` shouldn't handle this new fade method.
And I didn't realize there could be a different behavior if the sound is playing or not. Do you think that a fade action request on a sound not yet playing should be stored in memory and then used when the sound starts playing?

Reply:
Yes, as that is what the setVolume action does too. In fact, I am surprised that not all properties are handled like that, and only a select few.